src/log.h

```diff
@@ -84,14 +87,14 @@ class Logger {
 	const std::string getThreadName();
 
 	std::vector<ILogOutput *> m_outputs[LL_MAX];
+	std::atomic<bool> m_has_outputs[LL_MAX];
```

Comment:
how does this differ from `m_silenced_levels`?

Reply:
m_has_outputs tracks m_outputs. It's intended as a lock-free way of determining whether m_outputs[level] is empty.

m_silenced_levels is an externally set flag that can be turned on/off without affecting the contents of m_outputs.

It seems only be used in one place in the unit tests:

https://github.com/minetest/minetest/blob/master/src/unittest/test.cpp#L267

It might make sense to change this to remove the output instead.

Reply:
Not sure, can be kept like this.